presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -803,6 +805,22 @@ protected Type visitFunctionCall(FunctionCall node, StackableAstVisitorContext<C
                 throw e;
             }
 
+            if (node.getSortItems().isPresent()) {
```

Comment:
AggregationAnalyzer is only run for aggregation functions. So the check for whether the function is scalar or aggregation needs to happen somewhere else. I don't know whether this is the best place or not.